### PR TITLE
Add single category list type

### PR DESCRIPTION
**Pull request by alxy**
This adds the ability to also display single category links in a menu. It is heavily copied from the blog plugin, but does not support nested (child) cateogories. It will only display the one selected category.

### CODE REVIEW

File: models/Category.php [hunk 1]
```diff
@@ -154,36 +181,55 @@ public static function getMenuTypeInfo($type)
...
-
-                if ($item->children) {
-                    $branchItem['items'] = $iterator($item->children, $item->slug);
+            $entryUrl = $controller->pageUrl($pageUrl, ['slug' => $category->slug]);
+
+            $structure['url'] = $entryUrl;
+            $structure['isActive'] = $entryUrl === $url;
+            $structure['mtime'] = $category->updated_at;
+            $structure['title'] = $category->name;
+            $structure['code'] = $category->code;
```
Comment (tobias-kuendig) [inline models/Category.php]: Can you DRY this part up by extracting a `getMenuItem($url)` method?

```

Comment (alxy) [reply models/Category.php]: I decided to chose the signature of this method a little bit, but requested changes should be done anyway.

File: models/Category.php [hunk 2]
```diff
@@ -154,40 +187,69 @@ public static function getMenuTypeInfo($type)
...
+    protected static function getMenuItem($item, $url)
+    {
+        if ( ! $pageUrl = GeneralSettings::get('category_page')) {
+            throw new InvalidArgumentException(
+                'Mall: Please select a category page via the backend settings.'
+            );
+        }
+
+        $controller = new Controller();
+        $entryUrl = $controller->pageUrl($pageUrl, ['slug' => $item->slug]);
```
Comment (tobias-kuendig) [inline models/Category.php]: Here `$item->nestedSlug` should be used (this generates `/parent-1/parent-2/category` slugs). 

This should work with your single category items as well.

File: models/Category.php [hunk 3]
```diff
@@ -154,40 +187,69 @@ public static function getMenuTypeInfo($type)
...
+     * Creates a single menu item result array
+     *
+     * @param $item Category
+     * @param $url string
+     *
+     * @return array
+     */
+    protected static function getMenuItem($item, $url)
+    {
+        if ( ! $pageUrl = GeneralSettings::get('category_page')) {
```
Comment (tobias-kuendig) [inline models/Category.php]: Why did you move this guard here? Now it gets called once for every menu item. Can you move this back to `resolveMenuItem`?

Comment (tobias-kuendig) [reply models/Category.php]: No, sounds good to me!

File: models/Category.php (hunk 2 above)
Comment (alxy) [reply models/Category.php]: Fixed.

**Comment by tobias-kuendig** [comment]
Perfect! Great addition, thank you!
